crates/runtime/src/lib.rs

```diff
@@ -21,13 +21,17 @@ use std::collections::HashSet;
 use std::net::SocketAddr;
 use std::{collections::HashMap, sync::Arc};
 
+#[cfg(feature = "spice-metrics")]
+use ::datafusion::error::DataFusionError;
 use ::datafusion::sql::parser::{self, DFParser};
 use ::datafusion::sql::sqlparser::ast::{SetExpr, TableFactor};
 use ::datafusion::sql::sqlparser::dialect::PostgreSqlDialect;
 use ::datafusion::sql::sqlparser::{self, ast};
 use accelerated_table::AcceleratedTable;
 use app::App;
 use config::Config;
+#[cfg(feature = "spice-metrics")]
+use metrics::SetRecorderError;
```

Comment:
Not a huge fan of all these `#[cfg(feature = "spice-metrics")]` everywhere. Can we put most of the logic in a separate module that is conditionally compiled in? The module can live in this file, but just for better separation of the cfg.

Reply:
removed feature flag, that will be handled by spiced argument